### DISCUSSION TITLE
Multiclass data

### DISCUSSION THREAD

**Discussion by vee3capp** [mlr-org/mlr#2820] [Q&A]
I am trying to use the makeClassifTask, but my target is nonbinary. I have six stages I want to try and predict bacteria taxa for. Is there a way to set up the makeClassifTask function to have more than one target?

**Comment by pat-s** [comment]
Did you find the docs page about [multiclass classification](https://mlr.mlr-org.com/articles/tutorial/cost_sensitive_classif.html#multi-class-problems)? I am not an expert in this topic but there should be more resources to find (issues, stackoverflow) which might help.

PS: I suggest not to use `mlr` anymore but to start with `mlr3`.

**Reply by vee3capp** [reply]
Does mlr3 have all of the same functions? I am using mlr because it is what a workshop used that I attended. I am currently still stuck on the makeClassifTask classification. I have figured out kind of how to do the multi-task. When I run this code, 

asv_relative_abundance <- otu_table(phyloseq_relative_abundance)
asv_relative_abundance<-t(asv_relative_abundance)

metadata <- sample_data(phyloseq_relative_abundance)

model_decomp<- as.data.frame(cbind(asv_relative_abundance, "decomp.stage" = metadata$decomp.stage))

model_decomp$decomp.stage <- as.factor(model_decomp$decomp.stage )#Change the feature as 'factor'

trainTask<- makeClassifTask(data = model_decomp, target = "decomp.stage")

I get this error, 
Error in makeTask(type = type, data = data, weights = weights, blocking = blocking,  : 
  Assertion on 'data' failed: Must have colnames according to R's variable naming conventions, but element 1 does not comply.

I have checked the sample data/ code with the link you gave and my data input is set up exactly the same way. The only difference is that my classes are factors (for examge stage a, stage b, and so on)

**Reply by pat-s** [reply]
mlr3 should have the same functionality, if not, please open an issue and request it in https://github.com/mlr-org/mlr3 :)

`mlr` should not be used anymore for new projects, the learners were not updated and there might be uncaught issues meanwhile that might affect your outcome.

WRT to your issue: Please provide a [reproducible example](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example), preferably using the [reprex](https://github.com/tidyverse/reprex) package.
